crypto/fipsmodule/sha/internal.h

```diff
@@ -406,39 +409,91 @@ OPENSSL_EXPORT uint8_t *SHAKE128(const uint8_t *data, const size_t in_len,
 OPENSSL_EXPORT uint8_t *SHAKE256(const uint8_t *data, const size_t in_len,
                                  uint8_t *out, size_t out_len);
 
-// SHA3_Init initialises |ctx| fields through |FIPS202_Init| and 
-// returns 1 on success and 0 on failure.
+// SHA3_Init initialises |ctx| fields through |FIPS202_Init| and
+// returns 1 on success and 0 on failure. SHA3_Init fails if
+// |ctx| is nullptr or |bitlen| is inproper for SHA3 (any security
+// level) block size length. Otherwise, it propagates the |FIPS202_Init|
```

Comment:
As a caller of this function, the details of which functions are called underneath are not important -- but the caller needs to understand failure conditions. Here, one could recursively lookup the documentation of `FIPS202_Update` to understand its failure conditions, but it's a bit cumbersome. Is there a way one can express success/failure conditions at this level without resorting to implementation details?

Similar questions below.

Reply:
I updated the description https://github.com/aws/aws-lc/blob/026f32ed4a328356aca90712bd47ac459310fa97/crypto/fipsmodule/sha/internal.h#L422 following the suggested documentation format. Thank you.